### PR TITLE
[geometry] Test cloning of geometry interfaces

### PR DESCRIPTION
**Pull request by zcorpan**
Follows https://github.com/w3c/fxtf-drafts/pull/147

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/231129098)

# Firefox (nightly) #
Testing web-platform-tests at revision 822f748f4d91ddad8c7838a28c07ff121a1aa910
Using browser at version BuildID 20170511100157; SourceStamp d8762cb967423618ff0a488f14745f60964e5c49
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5888/css/geometry-1/cloning.html">/css/geometry-1/cloning.html</a></summary>


|                      Subtest                       | Results |                                                 Messages                                                 |
|----------------------------------------------------|---------|----------------------------------------------------------------------------------------------------------|
|                                                    | OK      |                                                                                                          |
| `DOMPointReadOnly clone: basic`                    | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMPointReadOnly clone: custom property`          | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMPointReadOnly clone: throwing getters`         | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMPointReadOnly clone: non-initial values`       | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMPoint clone: basic`                            | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMPoint clone: custom property`                  | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMPoint clone: throwing getters`                 | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMPoint clone: non-initial values`               | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMRectReadOnly clone: basic`                     | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMRectReadOnly clone: custom property`           | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMRectReadOnly clone: throwing getters`          | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMRectReadOnly clone: non-initial values`        | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMRect clone: basic`                             | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMRect clone: custom property`                   | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMRect clone: throwing getters`                  | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMRect clone: non-initial values`                | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMQuad clone: basic`                             | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMQuad clone: custom property`                   | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMQuad clone: throwing getters`                  | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMQuad clone: non-initial values`                | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMMatrixReadOnly clone: basic`                   | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMMatrixReadOnly clone: custom property`         | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMMatrixReadOnly clone: throwing getters`        | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMMatrixReadOnly clone: non-initial values (2d)` | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMMatrixReadOnly clone: non-initial values (3d)` | FAIL    | `Illegal constructor.`                                                                                   |
| `DOMMatrix clone: basic`                           | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMMatrix clone: custom property`                 | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMMatrix clone: throwing getters`                | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMMatrix clone: non-initial values (2d)`         | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |
| `DOMMatrix clone: non-initial values (3d)`         | FAIL    | `promise_test: Unhandled rejection with value: object "DataCloneError: The object could not be cloned."` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/231129104)

# Chrome (unstable) #
Testing web-platform-tests at revision 822f748f4d91ddad8c7838a28c07ff121a1aa910

**Comment by zcorpan** [comment]
Neither of Gecko or Chromium support cloning of geometry interfaces yet.

File: css/geometry-1/structured-serialization.html [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    object.foo = "bar";
+    return clone(object).then(other => {
+      assert_false("foo" in other, 'foo should not exist in other');
+      t.done();
+    });
+  }, `${prefix} custom property`);
+
+  promise_test(t => {
+    const object = new self[constr]();
+    defineThrowingGetters(object, attrs, 'should not be copied');
```
Comment (domenic) [inline css/geometry-1/structured-serialization.html]: Maybe "should not be called"

File: css/geometry-1/structured-serialization.html [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    return clone(object).then(other => {
+      for (let attr of attrs) {
+        other[attr];
```
Comment (domenic) [inline css/geometry-1/structured-serialization.html]: Maybe comment this line to explain what it's doing

**Review by domenic — APPROVED** [review]
Amazing tests!! I hope future structured-serialize tests are as comprehensive as this.

File: css/geometry-1/structured-serialization.html (hunk 1 above)
Comment (zcorpan) [inline css/geometry-1/structured-serialization.html]: I mean that the custom getter should not be copied to *other*. But I guess this is more confusing than helpful, so let's remove it, or move to a JS comment...

File: css/geometry-1/structured-serialization.html (hunk 2 above)
Comment (zcorpan) [inline css/geometry-1/structured-serialization.html]: Added comments.

**Pull request merged**